devops/lib/component.py

```diff
@@ -84,20 +84,29 @@ def validate(self, ctx=None):
             if result.returncode > 0:
                 raise ValidationError(f"Validation failed for {path}")
 
-    def build(self, ctx: Context, dry_run=False):
+    def build(self, ctx: Context, dry_run=False, docker_args=None):
         label(logger.info, f"Building {self.path}")
         dockerfile = self.path / "Dockerfile"
 
         if not dockerfile.exists():
             logger.info(f"No Dockerfile for {self.name} component")
             return
 
+        if isinstance(docker_args, list):
+            # Insert --build-arg before each item in docker_args.
+            docker_args = [["--build-arg", x] if x else "" for x in docker_args]
```

Comment:
Why is this `if x` required?

Reply:
Could you use a better name than `x`?

Reply:
It is required in case `--build-arg` is passed with no value, then run() will fail.  I wasn't sure how much of validation should be implemented.

E.g.

```
args = ["foo=bar", "", "hello=world"]
args = [["--build-arg", arg] for arg in args]
b = [arg for build_args_pair in args for arg in build_args_pair]

print(b) # ['--build-arg', 'foo=bar', '--build-arg', '', '--build-arg', 'hello=world']
```
Passing `*['--build-arg', 'foo=bar', '--build-arg', '', '--build-arg', 'hello=world']` to `run` command will fail.

Reply:
I don't think you understand the question - when is anyone going to pass `""` as an argument and we need to care about supporting it?

Reply:
Yeah if initial `build-args` will be coming also from other source than CLI, then it will be more suitable to preprocess `docker_args ` items then. But not now. Removed.